client/src/components/Form/Elements/FormSelect.vue

```diff
@@ -0,0 +1,215 @@
+                formattedOptions[i] = {
+                    label: this.options[i][0],
+                    value: this.options[i][1],
+                    default: this.options[i][2],
+                };
+            }
+            if (this.multiple == false && this.optional == true) {
+                formattedOptions.unshift({
+                    label: "Nothing selected",
+                    value: null,
+                    default: false,
+                });
+            }
+            return formattedOptions;
+        },
+        currentValue: {
+            get() {
+                // My understanding of this is that if we start with a value prop, use it.
+                // If there's a defaultValue set, find it in the array and use that.
+                // If there's no default value set, use the first item in the array with 'default' set to true.
+                // Lastly, just default to the first element in the array.
+                if (this.value == null) {
+                    return;
+                } else if (this.value !== "") {
+                    if (this.multiple) {
+                        return this.selectValueMultiple(this.value);
+                    } else if (this.display == "checkboxes") {
+                        return this.selectValueCheckboxes(this.value);
+                    } else if (this.display == "radio") {
+                        return this.selectValueSingle(this.value);
+                    } else {
+                        return this.selectValue(this.value);
+                    }
+                } else if (this.defaultValue !== "") {
+                    if (this.multiple || this.display == "checkboxes") {
+                        return this.selectValueMultiple(this.defaultValue);
+                    } else {
+                        return this.selectValue(this.defaultValue);
+                    }
+                } else {
+                    return this.selectDefaultLabelValue();
+                }
+            },
+            set(val) {
+                // Checkbox is mutliple, but not automatically set. If it IS set, this prevents the multiple from overriding the checkbox
+                if (val == null) {
+                    // This case can occur when single-select dropdown is re-selected
+                    return;
+                } else if (this.multiple && this.display != "checkboxes") {
+                    const values = this.getValuesFromFormattedOptions(val);
+                    this.$emit("input", values);
+                } else if (this.display == "radio" || this.display == "checkboxes") {
+                    this.$emit("input", val);
+                } else {
+                    this.$emit("input", val.value);
+                }
+            },
+        },
+    },
+    methods: {
+        selectValueMultiple(val) {
+            const selected = [];
+            const selectedValues = this.normalizeSelectedValues(val);
+
```

Comment:
Similar to the previous PR, recommending to remove the spaces from these functions, particularly if they are small. Also below please use `forEach` or even map reduce.

Reply:
Good idea to condense the logic, remove the loops. Change made.